### PR TITLE
【Paddle.Fleet】Fix tensor table

### PR DESCRIPTION
**Pull request by MrChengmo**
### PR types

New features,Bug fixes

### PR changes

Others

### Describe

在新的统一PS下完善了TensorTable
- 支持Server以Tensor存储参数
- 支持Server运行Program
- 支持PS各个Table独立的LearningRate Decay策略，目前支持动转静的4种decay策略：`'NoamDecay', 'NaturalExpDecay', 'InverseTimeDecay', 'ExponentialDecay'`

### CODE REVIEW

File: paddle/fluid/distributed/communicator_common.h [hunk 1]
```diff
@@ -30,7 +30,8 @@ struct CommContext {
               const std::vector<int64_t> &sections,
               const std::vector<std::string> &origin_names, int id,
               bool merge_add_ = true, bool is_sparse_ = true,
-              bool is_distributed_ = false, int table_id_ = -1)
+              bool is_distributed_ = false, int table_id_ = -1,
+              bool is_tensor_table_ = false)
```
Comment (seiriosPlus) [inline paddle/fluid/distributed/communicator_common.h:34]: 为什么要加 is_tensor_table_ 这个属性？

Comment (MrChengmo) [reply paddle/fluid/distributed/communicator_common.h:34]: - 在communicator 的 SendByCommunicator 中，需要根据send_ctx的不同类型进不同的发送策略，之前有dense/sparse，现在需要区分tensor_table，发送 global_step或其他feed_vars给server
- 在the_one_ps、public中也有类似的区分sparse/dense进不同table处理逻辑的部分，为了和dense/sparse table区分，加上了这个属性

**Review by seiriosPlus — APPROVED** [review]
LGTM， still need improvement.

**Review by luotao1 — APPROVED** [review]
LGTM for const_cast

File: python/paddle/fluid/incubate/fleet/parameter_server/ir/public.py [hunk 2]
```diff
@@ -1118,6 +1151,89 @@ def _get_optimize_ops(_program):
     return opt_ops
 
 
+def _add_lr_decay_table_pass(main_program, compiled_config, lr_decay_steps):
```
Comment (zhwesky2010) [inline python/paddle/fluid/incubate/fleet/parameter_server/ir/public.py:1154]: If other decay is needed? 

how about Use a map to convert `LRScheduler` to `Variable`   to reduce code

Comment (MrChengmo) [reply python/paddle/fluid/incubate/fleet/parameter_server/ir/public.py:1154]: After simple verification, these four methods can be directly supported at present, and new methods will be added according to user requirements in the future.

This code will be continuously optimized in the future, thx~

Comment (zhwesky2010) [reply python/paddle/fluid/incubate/fleet/parameter_server/ir/public.py:1154]: Ok , I think it can reuse code by dict,  too much if-else is not necessary.

**Review by zhwesky2010 — APPROVED** [review]
LGTM for lr decay

**Review by raindrops2sea — APPROVED** [review]

**Review by raindrops2sea — APPROVED** [review]

**Review by chalsliu — APPROVED** [review]

**Pull request merged**